pygbe/main.py

```diff
@@ -355,14 +367,27 @@ def main(argv=sys.argv, log_output=True, return_output_fname=False,
     print('-'*30)
     print('Total setup time   : {}s'.format(setup_time))
 
-    tic = time.time()
+
+    #   Check if there is a complex dielectric
+    complexDiel = 0
+    for f in field_array:
+        if numpy.iscomplexobj(f.E):
+            complexDiel = 1
+
 
     ### Solve
+    tic = time.time()
+
     print('Solve')
-    phi = numpy.zeros(param.Neq)
+    # Initializing phi dtype according to the problem we are solving.
+    if complexDiel == 1:
+        phi = numpy.zeros(param.Neq, type(f.E))
```

Comment:
Got it, though I'm not sure what is the best way of fix this. I don't think we want to save `f.E` just to check the types. Besides the difference between the 2 options is 1 line. If we want to use the `numpy.any()` we will need also to define, some way (not sure what's best) the `f.E`.